### PR TITLE
Print images to base64 (behind flag)

### PR DESCRIPTION
**Pull request by ericyd**
Add new setting to optionally print images to base64 when rendering HTML (defaults to false).

Closes [73](https://github.com/neilsustc/vscode-markdown/issues/73).

In case the author includes an invalid file path in their MD doc, a warning notification will display and the extension will fall back to the original behavior.

### CODE REVIEW

File: src/print.ts [hunk 1]
```diff
@@ -63,6 +63,19 @@ function print(type: string) {
 
     if (vscode.workspace.getConfiguration("markdown.extension.print", doc.uri).get<boolean>("absoluteImgPath")) {
         body = body.replace(/(<img[^>]+src=")([^"]+)("[^>]*>)/g, function (match, p1, p2, p3) { // Match '<img...src="..."...>'
+            if (vscode.workspace.getConfiguration("markdown.extension.print", doc.uri).get<boolean>("imgToBase64")) {
+                try {
+                    let root;
+                    const imgPath = (root = vscode.workspace.getWorkspaceFolder(doc.uri))
+                        ? path.join(root.uri.fsPath, p2)
+                        : path.join(path.dirname(doc.uri.fsPath), p2);
```
Comment (neilsustc) [inline src/print.ts]: How about changing the `fixHref` function to return an `Uri` so that we can reuse it here (`imgPath = fixHref(...).fsPath`).

**Comment by ericyd** [comment]
That's a great idea. Let me know if this is what you had in mind.

File: src/print.ts [hunk 2]
```diff
@@ -145,7 +142,7 @@ function getCustomStyleSheets(resource: vscode.Uri): string[] {
     const styles = vscode.workspace.getConfiguration('markdown')['styles'];
     if (styles && Array.isArray(styles) && styles.length > 0) {
         return styles.map(s => {
-            let uri = vscode.Uri.parse(fixHref(resource, s));
+            let uri = vscode.Uri.parse(fixHref(resource, s).toString());
```
Comment (neilsustc) [inline src/print.ts]: 😂

Comment (ericyd) [reply src/print.ts]: haha, refactoring without coffee :)

**Comment by ericyd** [comment]
Sounds great, ready for review

**Comment by neilsustc** [comment]
🍻

**Pull request merged**